homeassistant/components/media_player/braviatv.py

```diff
@@ -0,0 +1,473 @@
+                                            pin,
+                                            CLIENTID_PREFIX,
+                                            NICKNAME)
+        if not cookie:
+            self._state = STATE_OFF
+            return
+        else:
+            self._cookies = cookie
+            # update the state first of all
+            self.update()
+
+    def update(self):
+        """Update TV info."""
+        if self._cookies is None:
+            cookie = self._braviarc.bravia_auth(self._host,
+                                                self._pin,
+                                                CLIENTID_PREFIX,
+                                                NICKNAME)
+            if not cookie:
+                return
+            else:
+                self._cookies = cookie
+
+        # Retrieve the latest data.
+        try:
+            resp = self._braviarc.\
+                bravia_req_json(self._host,
+                                self._cookies,
+                                "sony/avContent",
+                                _jdata_build(
+                                    "getPlayingContentInfo",
+                                    None))
+            if resp is None:
+                self._state = STATE_OFF
+            elif not resp.get('error'):
+                self._state = STATE_ON
+                playing_content_data = resp.get('result')[0]
+                self._program_name = playing_content_data.get('programTitle')
+                self._channel_name = playing_content_data.get('title')
+                self._program_media_type = playing_content_data.get(
+                    'programMediaType')
+                self._channel_number = playing_content_data.get('dispNum')
+                self._source = playing_content_data.get('source')
+                self._content_uri = playing_content_data.get('uri')
+                self._duration = playing_content_data.get('durationSec')
+                self._start_date_time = playing_content_data.get(
+                    'startDateTime')
+
+                # refresh volume info:
+                self._refresh_volume()
+
+                # update command data the very first time
+                if len(self._commands) == 0:
+                    self._refresh_commands()
+
+                if len(self._source_list) == 0:
+                    self._content_mapping = self._braviarc.\
+                        load_source_list(self._host, self._cookies)
+                    self._source_list = []
+                    for key in self._content_mapping:
+                        self._source_list.append(key)
+
+            else:
+                self._state = STATE_OFF
+
+        except Exception as exception_instance:  # pylint: disable=broad-except
+            _LOGGER.error(exception_instance)
+            self._state = STATE_OFF
+
+    def _refresh_volume(self):
+        resp = self. \
+            _braviarc.bravia_req_json(self._host,
```

Comment:
Wouldn't it make sense to move such calls also to the Bravia package? That way you can call things like

```python
resp = self._braviarc.volume_info()
resp = self._braviarc.remote_controller_info()
# etc…
```